modules/javafx.graphics/src/main/java/javafx/css/CssMetaData.java

```diff
@@ -327,5 +328,23 @@ public int hashCode() {
             .append("}").toString();
     }
 
-
+    /**
+     * Utility method which combines {@code CssMetaData} items in one unmodifiable list with size equal
+     * to the number of items it holds.
+     *
+     * @param list the css metadata items, usually from the parent, must not be null
+     * @param items the additional items
+     * @return the unmodifiable list containing all of the items
+     *
+     * @since 22
+     */
+    public static List<CssMetaData<? extends Styleable, ?>> combine(
+        List<CssMetaData<? extends Styleable, ?>> list,
```

Comment:
Any reason this is specifically a `List`? Can it not be a `Collection`?

Reply:
It should have been a Set maybe, or indeed a Collection.

But it is too late: Node codifies the List return values:

```
public static List<CssMetaData<? extends Styleable, ?>> getClassCssMetaData();

public List<CssMetaData<? extends Styleable, ?>> getCssMetaData();
```

Reply:
Nir was probably referring to the type of the `list` parameter.
As for the return value, not only must this be a `List`, it should also be mandated to be `RandomAccess`. Any implementation that doesn't support random access may cause severe performance degradation.